corehq/messaging/scheduling/views.py

```diff
@@ -107,6 +117,130 @@ def get(self, request, *args, **kwargs):
+        return MessageForm(**self.form_kwargs)
+
+    @property
+    def page_context(self):
+        return {
+            'form': self.message_form,
+        }
+
+    @cached_property
+    def project_languages(self):
+        doc = StandaloneTranslationDoc.get_obj(self.domain, 'sms')
+        return getattr(doc, 'langs', ['en'])
+
+    def post(self, request, *args, **kwargs):
+        if self.message_form.is_valid():
+            # TODO editing should not create a new one
+            values = self.message_form.cleaned_data
+            if values['send_frequency'] == 'immediately':
+                if values['translate']:
+                    messages = {}
+                    for lang in self.project_languages:
+                        key = 'message_%s' % lang
+                        if key in values:
+                            messages[lang] = values[key]
+                    content = SMSContent(message=messages)
+                else:
+                    content = SMSContent(message={'*': values['non_translated_message']})
+                schedule = AlertSchedule.create_simple_alert(self.domain, content)
+                broadcast = ImmediateBroadcast(
+                    domain=self.domain, name=values['schedule_name'], schedule=schedule
+                )
+                broadcast.save()
+                recipients = [('CommCareUser', r_id) for r_id in values['recipients']]
+                refresh_alert_schedule_instances.delay(schedule, recipients)
+
+            return HttpResponseRedirect(reverse(BroadcastListView.urlname, args=[self.domain]))
+        return self.get(request, *args, **kwargs)
+
+
+class EditMessageView(CreateMessageView):
+    urlname = 'edit_message'
+    page_title = ugettext_lazy('Edit Message')
+
+    @property
+    def page_url(self):
+        return reverse(self.urlname, args=[self.domain, self.broadcast_id])
+
+    @property
+    def broadcast_id(self):
+        return self.kwargs.get('broadcast_id')
+
+    @cached_property
+    def broadcast(self):
+        try:
+            broadcast = ImmediateBroadcast.objects.prefetch_related('schedule').get(pk=self.broadcast_id)
+        except:
+            raise Http404()
+
+        return broadcast
+
+    @cached_property
+    def message_form(self):
+        if self.request.method == 'POST':
+            return MessageForm(self.request.POST, **self.form_kwargs)
+
+        broadcast = self.broadcast
+        schedule = broadcast.schedule
+        schedule_instances = get_alert_schedule_instances_for_schedule(schedule)
+        recipients = [
+            (instance.recipient_type, instance.recipient_id)
+            for instance in schedule_instances
+        ]
+        initial = {
+            'schedule_name': broadcast.name,
+            'send_frequency': 'immediately',
+            'recipients': recipients,
+            'content': 'sms',
+            'message': schedule.memoized_events[0].content.message,
```

Comment:
this will only work if content is an instance of `SMSContent`

Reply:
I added a comment https://github.com/dimagi/commcare-hq/pull/17722/commits/d2a7d2bc780416e0d7463b42b5a52d77f8094ab4. I plan on handling this when adding support for surveys and emails